### PR TITLE
Add the `code tab` to the creator page.

### PR DESCRIPTION
**Pull request by Aritra8438**
This PR adds the code tab to the creator page so that users can write codes for the problems they are creating.

https://github.com/omegaup/omegaup/assets/64671908/a3bdcca3-bcaf-4899-8007-67517d6b07f7

Fixes: #7593

# Comments
- There is an option to upload code files.
- After uploading, the language will automatically be changed inferring the file language from the extension.
- After clicking `save`, the code and the extension will be saved to the creator store. 

# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [x] The tests were executed and all of them passed.
- [x] If you are creating a feature, the new tests were added.
- [ ] If the change is large (> 200 lines), this PR was split into various Pull Requests. It's preferred to create one PR for changes in controllers + unit tests in PHPUnit,  and then another Pull Request for UI + tests in Jest, Cypress or both.

### CODE REVIEW

**Comment by Aritra8438** [comment]
Hey @carlosabcs, PTAL!
I will soon add the tests for this in the same PR.

Can we divide this PR? It's a bit hard as everything is linked and relevant to each other.

File: frontend/www/js/omegaup/components/arena/CodeView.vue [hunk 1]
```diff
@@ -117,6 +117,7 @@ export default class CodeView extends Vue {
     this.mode = languageModeMap[newLanguage];
   }
   mounted() {
+    this.refresh();
```
Comment (carlosabcs) [inline frontend/www/js/omegaup/components/arena/CodeView.vue:120]: Why is this necessary?

File: frontend/www/js/omegaup/components/problem/creator/code/CodeTab.test.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    const mockReadFileMethod = jest
+      .spyOn(wrapper.vm, 'readFile')
+      .mockImplementation(() => testFile);
+
+    const fileInput = wrapper.find('input[type=file]');
+    await fileInput.trigger('change');
+
+    expect(mockReadFileMethod).toHaveBeenCalled();
+
+    await new Promise((r) => setTimeout(r, 1000)); // Waiting for the FileReader to complete reading the file.
```
Comment (carlosabcs) [inline frontend/www/js/omegaup/components/problem/creator/code/CodeTab.test.ts:31]: Can you please provide a better explanation for this?

Comment (Aritra8438) [reply frontend/www/js/omegaup/components/problem/creator/code/CodeTab.test.ts:31]: Sure @carlosabcs. As mentioned [here](https://blog.shovonhasan.com/using-promises-with-filereader/#:~:text=the%20filereader%20methods%20work%20asynchronously%20but%20don't%20return%20a%20promise.%20and%20attempting%20to%20retrieve%20the%20result%20immediately%20after%20calling%20a%20method%20will%20not%20work%2C%20as%20the%20.onload%20event%20handler%20fires%20only%20after%20the%20filereader%20has%20successfully%20finished%20reading%20the%20file%20and%20updates%20the%20filereader's%20), The FileReader methods work asynchronously but don't return a Promise. And attempting to retrieve the result immediately after calling a method will not work, as the .onload event handler fires only after the FileReader has successfully finished reading the file.

As a [workaround](https://stackoverflow.com/questions/51026420/filereader-readastext-async-issues), we can wrap the whole `FileReader` reading the process into a `promise`.

File: frontend/www/js/omegaup/components/arena/CodeView.vue (hunk 1 above)
Comment (Aritra8438) [reply frontend/www/js/omegaup/components/arena/CodeView.vue:120]: The issue I faced was [this](https://stackoverflow.com/questions/58951737/codemirror-line-numbers-overlapped-with-code).


https://github.com/omegaup/omegaup/assets/64671908/5b4eb7bb-1382-4680-a877-f286fb237fcb



As a solution, I refreshed the editor on being mounted as mentioned in the so blog

Comment (carlosabcs) [reply frontend/www/js/omegaup/components/arena/CodeView.vue:120]: Ok, that seems acceptable to me. Do you have other opinions @tvanessa, @pabo99, @heduenas ?

File: frontend/www/js/omegaup/components/problem/creator/code/CodeTab.test.ts (hunk 2 above)
Comment (carlosabcs) [reply frontend/www/js/omegaup/components/problem/creator/code/CodeTab.test.ts:31]: Ok, now this is clear to me.

**Review by carlosabcs — APPROVED** [review]

**Review by heduenas — APPROVED** [review]
